woocommerce/wc-admin
# Refactoring WCPayCard into new onboarding package

Fixes woocommerce/woocommerce-admin#6868

Creates a new package, `@woocommerce/tasks` to house the refactored WCPayCard components. These will be consumed by the WCPay repo, as well as used on the payments tasks page as it was previously. They've been refactored to aid in flexibility for how they're consumed. 

I've also moved the slotfill components into this package, so we'll likely need to publish this for the `remote-payments-tester` to work properly. The alternative is using `npm link` in the short-term.

Also included are other revisions and tweaks to the logic around `wcpay` to ensure that its working as expected.

_Other Note:_ There was a little discussion on how "broad" to make this new package, with talk of having a more general one around `onboarding` in total. After Sam's [recent p2 post](https://teamisotope.wordpress.com/2021/05/13/woocommerce-components-is-too-big/) I thought something a bit smaller and more granular might better suit the current trend here.

### Open questions
- I initially had the new tasks package added to [the DEWP config](https://github.com/woocommerce/woocommerce-admin/blob/main/packages/dependency-extraction-webpack-plugin/assets/packages.js), but I eventually figured out is was causing an issue and the react app wasn't loading at all. I can revisit, but wasn't sure if this was necessary for this package?

### Screenshots

![image](https://user-images.githubusercontent.com/444632/119566468-08f95480-bd60-11eb-84c3-82959ae12320.png)


### Detailed test instructions:

-  Enable remote payments and ensure your opted into marketing
-  Clone the `try/use-tasks-package` branch of [my fork of the remote payment tester](https://github.com/joelclimbsthings/woocommerce-admin-remote-tester)
- Set your store to a US based location
- Use `yarn link` to link your remote payment tester to the new `@woocommerce/tasks` package
- Ensure that the WCPay Card appears as pictured when you navigate to the payment task
- Click on "Get Started," and verify that the slot-fill works. Note that no fields have been setup, so it will just display the "Proceed" button for now.
